lib/trino-parquet/src/main/java/io/trino/parquet/reader/ParquetReader.java

```diff
@@ -415,21 +412,20 @@ private ColumnChunk readStruct(GroupField field)
         return new ColumnChunk(rowBlock, columnChunk.getDefinitionLevels(), columnChunk.getRepetitionLevels());
     }
 
-    private FilteredOffsetIndex getFilteredOffsetIndex(int rowGroup, long rowGroupRowCount, ColumnPath columnPath)
+    @Nullable
+    private FilteredOffsetIndex getFilteredOffsetIndex(FilteredRowRanges rowRanges, int rowGroup, long rowGroupRowCount, ColumnPath columnPath)
     {
-        if (filter.isPresent()) {
-            RowRanges rowRanges = getRowRanges(filter.get(), rowGroup);
-            if (rowRanges != null && rowRanges.rowCount() < rowGroupRowCount) {
-                Optional<ColumnIndexStore> columnIndexStore = this.columnIndexStore.get(rowGroup);
-                if (columnIndexStore.isPresent()) {
-                    OffsetIndex offsetIndex = columnIndexStore.get().getOffsetIndex(columnPath);
-                    if (offsetIndex != null) {
-                        return FilteredOffsetIndex.filterOffsetIndex(offsetIndex, rowRanges, rowGroupRowCount);
-                    }
-                }
-            }
+        if (rowRanges == null) {
+            return null;
         }
-        return null;
+        // rowRanges is non-null only when columnIndexStore is present for a rowGroup
```

Comment:
This comment doesn't belong here -- or rather, the fact that there has to be coordinated knowledge between this method and its caller is an indication that there something wrong in the abstraction. Ideally, `columnIndexStore.get(rowGroup)` should be done in one place and the result passed around as necessary.

Reply:
This was just taking advantage of the presence of `rowRanges` to avoid an if on the optional for row group column index. I've restored the `if` and gotten rid of the comment.